scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/FTPConnector.java

```diff
@@ -0,0 +1,390 @@
+
+    private String ftpLocalFilePath = null;
+
+    private String ftpRemoteFilePath = null;
+
+    private String ftpMode = null;
+
+    private String ftpUsername = null;
+
+    private String ftpPassword = null;
+
+    private static final String FTP_LOCAL_FILE_PATH = "ftpLocalFilePath";
+
+    /**
+     * @see JavaExecutable#init(Map)
+     */
+    @Override
+    public void init(Map<String, Serializable> args) throws Exception {
+        if (args.containsKey("ftpHostname")) {
+            ftpHostname = args.get("ftpHostname").toString();
+        }
+        if (args.containsKey("ftpPort")) {
+            try {
+                ftpPort = Integer.parseInt(args.get("ftpPort").toString());
+            } catch (NumberFormatException e) {
+            }
+        }
+        if (args.containsKey("ftpExtractArchive")) {
+            ftpExtractArchive = Boolean.parseBoolean(args.get("ftpExtractArchive").toString());
+        }
+        if (args.containsKey(FTP_LOCAL_FILE_PATH)) {
+            ftpLocalFilePath = args.get(FTP_LOCAL_FILE_PATH) == null ? "" : args.get(FTP_LOCAL_FILE_PATH).toString();
+        }
+        if (args.containsKey("ftpRemoteFilePath")) {
+            ftpRemoteFilePath = args.get("ftpRemoteFilePath").toString();
+        }
+        if (args.containsKey("ftpMode")) {
+            ftpMode = args.get("ftpMode").toString();
+        }
+        ftpUsername = getThirdPartyCredential("FTP_USERNAME");
+        ftpPassword = getThirdPartyCredential("FTP_PASSWORD");
+        if (ftpUsername == null || ftpPassword == null) {
+            throw new IllegalArgumentException("You first need to add your ftp username and password (FTP_USERNAME, FTP_PASSWORD) to the third party credentials");
+        }
+    }
+
+    /**
+     * @see JavaExecutable#execute(TaskResult[])
+     */
+    @Override
+    public Serializable execute(TaskResult... results) throws IOException, IllegalArgumentException {
+
+        FTPClient ftpClient = new FTPClient();
+        // connect and login to the server
+        ftpClient.connect(ftpHostname, ftpPort);
+        int reply = ftpClient.getReplyCode();
+        if (!FTPReply.isPositiveCompletion(reply)) {
+            ftpClient.disconnect();
+            throw new IOException("Exception in connecting to FTP Server");
+        }
+        ftpClient.login(ftpUsername, ftpPassword);
+
+        // use local passive mode to pass firewall
+        ftpClient.enterLocalPassiveMode();
+
+        getOut().println("Connected");
+
+        //FTP mode is GET
+        if (ftpMode.equals("GET")) {
+            FtpGet(ftpClient);
+        }
+        //FTP mode is PUT
+        else if (ftpMode.equals("PUT")) {
+            FtpPut(ftpClient);
+        }
+        // log out and disconnect from the server
+        ftpClient.logout();
+        ftpClient.disconnect();
+
+        getOut().println("Disconnected");
+
+        return 1;
+    }
+
+    private void FtpGet(FTPClient ftpClient) throws IOException, IllegalArgumentException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.lang.IllegalArgumentException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)

Reply:
Agree

Reply:
done